feat: set ServerBindAddr config item, fixes #93

charles-knox-intel opened the pull request:
Signed-off-by: charles-knox-intel <44684517+charles-knox-intel@users.noreply.github.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

<!-- If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/master/.github/CONTRIBUTING.md -->

## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior and link to a relevant issue. -->

All profiles in this repository currently only listen on `0.0.0.0`. The `app-functions-sdk-go` dependency does not currently provide a means of configuring this.

Issue Number: #93

## What is the new behavior?

The `ServerBindAddr` service configuration option has been added in `app-functions-sdk-go` (https://github.com/edgexfoundry/app-functions-sdk-go/pull/406) as well as `go-mod-bootstrap` (https://github.com/edgexfoundry/go-mod-bootstrap/pull/84). As requested in #93, the configuration option has been added to all services in this repository as well.

## Does this PR introduce a breaking change?
<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?

No

## Are there any specific instructions or things that should be known prior to reviewing?

This change relies on https://github.com/edgexfoundry/app-functions-sdk-go/pull/406 as a dependency. It may not have any effect until that change is merged, a release is created, and that release is added to this repository as a dependency.

## Other information

lenny-intel commented on res/blackbox-tests/configuration.toml:
```diff
@@ -69,6 +69,7 @@
 BootTimeout = "30s"
 CheckInterval = "10s"
 Host = "localhost"
+ServerBindAddr = ""
```
Add comment why left blank. See example in the issue. https://github.com/edgexfoundry/app-service-configurable/issues/93

charles-knox-intel commented:
Addressed PR feedback from @lenny-intel. Tested by running:

```bash
go build

EDGEX_SECURITY_SECRET_STORE=false SERVICE_SERVERBINDADDR=127.0.0.1 ./app-service-configurable -s -p sample
```
